### DISCUSSION TITLE
Why osu!catch needs an option for fruit rotation in skinning

### DISCUSSION THREAD

**Discussion by strngth** [ppy/osu#25103] [Ideas]
In this document, I explain why having the option to have fruit rotation in skinning is needed.
https://docs.google.com/document/d/1J5JlRfKnDlFy3z4R4hCe4gvcX1MtC_NeujAftQv4QB0/edit?usp=sharing

  If you come from twitter, please upvote and leave comments.

**Comment by bdach** [comment]
To my (admittedly naive) understanding this proposal has the potential of making a certain class of skin unanimously better to use than another. The thinking is that the optimal way to play using this would be to use a white square for the fruit sprite with rotation off, which would make it much easier to read the extent of the fruit hitbox.

Probably needs community discussion first, probably needs a skinning system in lazer that can affect gameplay elements (not hud which we have) to come online.

**Reply by strngth** [reply]
Hello, it is so nice to read your message and have some discussion in this thread. 

  Currently, the main metas for this game mode are actually 2 skins; circles, and squares. Actually, having fruit rotation rng actually does what you say you having no rng would do. 
  
  Let me explain.
  
  Since there is an RNG factor to rotation, I want to say about 80-90%+ of players use a circle skin, since rotation does not effect the shape. Squares are truly only used because players such as "ExGon" and "Motion" (former #1 catch players) both use this fruit shape. Having no fruit rotation would allow for so much more complex-looking fruit shapes that the skinning community could create.
  
  The downside to no fruit rotation is that a player can create a fruit image of 256x2000 and have a line coming down so that they can see a note before it appears in the flashlight as a skin exploit. You would have to cap the image size to a reasonable amount or else osu! would default back to normal fruit shapes.
  
  That is not to say there aren't already skin exploits in catch that effect flashlight, because "orbit" does exist. (a skin with a large ring around the fruits so players can see before fruit is in flashlight render)
  
  There are many skinning possibilities that don't just have to do with fruit rotations but, this would be a wonderful QoL change and I hope to see it in the game.

**Reply by strngth** [reply]
Shapes like a diamond, ovals, any polygon shape, characters(?), and much much more would all be far more viable. It would be a nice change for the current meta of 2 boring shapes that we've been using for over a decade.

**Reply by peppy** [reply]
If this is to even be considered, there needs to be some kind of forced shape "solid" factor.

osu!catch is *not* turning into a mania as people optimise skins down to horizontal line rectangular strips, which is all I see in this proposal.

**Reply by strngth** [reply]
I can understand your thought process behind a horizontal line strip shape. However, when you say, "osu!catch is not turning into a mania as people optimise skins down to horizontal line rectangular strips" it actually already has. 90% of players are using circles, circumventing the RNG that is in place currently. We have been doing so for over a decade at this point, myself included.
  
 I don't see an issue with giving someone an option to have a strip-shaped fruit skin, because not everyone will use this shape. Everyone will find their own shape that they like to play with, not limited by the RNG rotation factor.

  We are on the same page and in agreeance actually. I don't want to see everyone using the same exact skin, that's so boring. Let's change this so we see more variety in gameplay.

**Reply by peppy** [reply]
> I don't see an issue with giving someone an option to have a strip-shaped fruit skin, because not everyone will use this shape.

Until it become the new "pro meta" like circles due to offering minor readability advantages.

**Reply by strngth** [reply]
> Until it become the new "pro meta" like circles due to offering minor readability advantages.



I believe there are pro's and con's to any change. However, you are assuming or basing the opinion that the change is bad off of something that hasn't happened. There is a level of negative connotation for the sake of argument and ignoring many of the improvements this change will bring.

  I respect your thought process in considering possible disadvantages, if they even are disadvantages in the first place. Believe me, I have as well as stated in my first reply to @bdach . However, being open-minded about this change can open new possibilities for our community. 
  
  We haven't had any changes to visual customization since 2015, that being the inclusion of HyperDash, HyperDashAfterImage, HyperDashFruit modifiable RGB values in the config file. The state of our game / community has evolved quite substantially since then. We are held back, and in some sense, left behind in comparison to all the modifiable things you can do to standard, and mania. 
  
  Having this being a possibility is a step in the right direction, and I am happy to further our discussions with more changes to improve the game mode as a whole; not just with customization changes.

**Reply by DatDoddo** [reply]
As a player who loves to mess around with skin elements, I've always found the randomly rotated fruits as well as the continuously rotating juice stream to be an odd choice, as they do nothing but hinder reading. In the same way, if this discussion were to spark change and the rotation from both the fruits and the juice stream is removed, using a non-symmetrical shape (both horizontally and vertically) would feel rather uncomfortable to read despite the reading advantage it may provide on paper due to density of the notes being lower. A vertically symmetrical shape would personally be a more comfortable shape to read, yet still a complete 180 from what we have now.

In my eyes, most players would continue using a circle skin as it is simply easier on the eyes. There's a reason why circle skins in 4k mania have gained a large user base and bar skins aren't the norm anymore, despite offering the same advantage as they would in this case. Thus, this change would only serve to add more variety to the current monopoly that circle skins have.

If the "pro meta" worries are the only ones stopping this from happening, is there a possibility to limit fruit elements to only be able to contain fully symmetrical shapes? And if that condition is not fulfilled the fruit element turns fully transparent during gameplay?

A question I'm wondering about is: wouldn't any change, no matter what it may be, have players try to find a meta? There's a meta right now and there would be one in the future as well.. So disabling rotation would truly only have the effect of adding variety to skinning. 
The only way to circumvent the meta chasing is to use a standardized skin and not allow for customization.

**Reply by Ajiaaaa** [reply]
I strongly believe that this proposal opens the door to creativity and variety, without truly endangering game balance.

Whether we allow rotation or not, there will always be a meta — t's already been circles for over a decade. Removing fruit rotation wouldn't necessarily "create" a meta — it would just shift the shape of it (literally).

I don’t think this should be a reason to hold the game back from what could be a major step forward for skinning in osu!catch. If anything, it would give more power to the community to express themselves with new fruit shapes beyond just circles and squares.

And as @DatDoddo said: readability and comfort will always matter more than theoretical advantages. Even if some shapes end up giving slight visibility benefits, players will still gravitate toward what feels natural to play with — and that diversity is a good thing.

**Comment by strngth** [comment]
So, where is reply? It's been a week at this point.
